### ISSUE TITLE
Warning: "circular require considered harmful", Ruby 2.2.0 with Faraday

### ISSUE THREAD

**Issue by nthx** [mashiro/saorin#2]
Hello there.

I've been using `saorin` together with `faraday` for a while.
Whenever I run `bundle exec rspec` I get bunch of warning on the console. Code behaves correctly though. Can you suggest what could be wrong? Is it something you can fix?

gemspec:
```
lib = File.expand_path('../lib', __FILE__)

Gem::Specification.new do |spec|
  #...
  spec.files         = `git ls-files`.split($/)
  spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
  spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
  spec.require_paths = ["lib"]

  spec.add_dependency "faraday", "~> 0.9" #saorin uses it, but doesn't require explicitly
  spec.add_dependency "saorin", "~> 0.6"

  spec.add_development_dependency "bundler", "~> 1.5"
  spec.add_development_dependency "rake"
  spec.add_development_dependency "rspec", "~> 3.0"
end
```

Warning message:
```
nthx@thepik:~/cleeng (master)$ bundle exec rspec
/home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.0/lib/saorin/request.rb:1: warning: loading in progress, circular require considered harmful - /home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.0/lib/saorin.rb
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in  `<main>'
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in  `eval'
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/rspec:23:in  `<main>'
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/rspec:23:in  `load'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/exe/rspec:4:in  `<top (required)>'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/runner.rb:38:in  `invoke'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/runner.rb:70:in  `run'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/runner.rb:85:in  `run'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/runner.rb:97:in  `setup'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/configuration.rb:1224:in  `load_spec_files'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/configuration.rb:1224:in  `each'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/configuration.rb:1226:in  `block in load_spec_files'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/configuration.rb:1226:in  `load'
	from /xxx/spec/acceptance/cleeng_spec.rb:1:in  `<top (required)>'
	from /xxx/spec/acceptance/cleeng_spec.rb:1:in  `require_relative'
	from /xxx/lib/cleeng.rb:2:in  `<top (required)>'
	from /xxx/lib/cleeng.rb:2:in  `require_relative'
	from /xxx/lib/cleeng/api.rb:1:in  `<top (required)>'
	from /xxx/lib/cleeng/api.rb:1:in  `require'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.0/lib/saorin.rb:3:in  `<top (required)>'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.0/lib/saorin.rb:3:in  `require'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.0/lib/saorin/request.rb:1:in  `<top (required)>'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.0/lib/saorin/request.rb:1:in  `require'
/home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.0/lib/saorin/response.rb:1: warning: loading in progress, circular require considered harmful - /home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.0/lib/saorin.rb
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in  `<main>'

```

And obvious "require" in the final code..

```ruby
require 'saorin'
#...
```

**Comment by mashiro** [comment]
Sorry, it is unnecessary require was caused.
Now fixed and released v0.6.1. Please try it.
Thanks.

**Comment by nthx** [comment]
Thanks for quick reaction. 
Nope. Still message is the same :-(

```
nthx@thepik:~/cleeng (master)$ bundle exec rspec
/home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.1/lib/saorin/server/base.rb:5: warning: loading in progress, circular require considered harmful - /home/nthx/.rvm/gems/ruby-2.2.0/gems/saorin-0.6.1/lib/saorin/server.rb
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in  `<main>'
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in  `eval'
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/rspec:23:in  `<main>'
	from /home/nthx/.rvm/gems/ruby-2.2.0/bin/rspec:23:in  `load'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/exe/rspec:4:in  `<top (required)>'
	from /home/nthx/.rvm/gems/ruby-2.2.0/gems/rspec-core-3.2.2/lib/rspec/core/runner.rb:38:in  `invoke'

```

**Comment by mashiro** [comment]
Oh, maybe fix leak.
Could you try to remove this line. https://github.com/mashiro/saorin/blob/master/lib/saorin/server/base.rb#L5

**Comment by mashiro** [comment]
OK, it was reproduce when run rspec with --warning option.
Problem does not occur if perhaps use the v0.6.2

**Comment by nthx** [comment]
+1

It's fixed in 0.6.2

Thanks a lot.
